how-to-guides/sagemaker/custom-docker-container/Using-Neptune-with-custom-container.ipynb

```diff
@@ -0,0 +1,268 @@
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "id": "cb87bfb5-8313-4f4a-9399-b5acfaba5df9",
+   "metadata": {},
+   "source": [
+    "### Training script\n",
+    "\n",
+    "The training script can be found in `decision_trees/train`. As compared to the script from AWS's tutorial, we added few lines of our code to it:\n",
+    "\n",
+    "```diff\n",
+    "[...]\n",
+    "\n",
+    "+ import neptune.new as neptune\n",
+    "+ import neptune.new.integrations.sklearn as npt_utils\n",
+    "\n",
+    "[...]\n",
+    "\n",
+    "def train():\n",
+    "+    run = neptune.init_run(\n",
+    "+         tags=[\"sagemaker\"],\n",
+    "+         source_files=[\"train\"],\n",
+    "+    )\n",
+    "     [...]\n",
+    "     \n",
+    "     # Now use scikit-learn's decision tree classifier to train the model.\n",
+    "     clf = tree.DecisionTreeClassifier(max_leaf_nodes=max_leaf_nodes)\n",
+    "     clf = clf.fit(train_X, train_y)\n",
+    "\n",
+    "+    run[\"cls_summary\"] = npt_utils.create_classifier_summary(\n",
+    "+       clf, train_X, train_X, train_y, train_y\n",
+    "+    )\n",
+    "     [...]\n",
+    "```"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "1351effd-f554-4621-b552-02bd273abb5e",
+   "metadata": {},
+   "source": [
+    "### Build the Docker image and push it to ECR\n",
+    "\n",
+    "Next, we need to build the container. The following Bash script assumes that you have [AWS CLI](https://aws.amazon.com/cli/) installed on your machine. It again uses the code from the AWS tutorial. It is lengthy, because it automatically creates an ECR repository for us and pushes the image to it."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "f6dfbd2e",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "!cat build_and_push.sh"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "e054e735",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "!bash ./build_and_push.sh"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "id": "e35d838f-b0e0-4491-9d71-7b52799e47a6",
+   "metadata": {},
+   "source": [
+    "## Start the training job\n",
+    "\n",
+    "<div class=\"alert alert-info\">This part of the notebook is to be run from the AWS SageMaker Notebook.</div>"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "4d635bbe-d2fe-453b-be68-96d301d3e8ee",
+   "metadata": {},
+   "source": [
+    "### Obtaining the Neptune token from AWS Secrets\n",
+    "\n",
+    "If you store Neptune token and project name in AWS Secrets, you can read them using the following code. If you do that, you can use your token and project name in the place of `NEPTUNE_API_TOKEN` and `NEPTUNE_PROJECT` in the next cell. Alternativelly, you can add the code below to the `decision_trees/train` script before building the Docker image and read the secrets from the `secrets` dictionary instead of `os.envir`.\n",
```

Comment:
```suggestion
    "If you store a Neptune API token and project name in AWS Secrets, you can read them using the following code. If you do that, you can use your token and project name in the place of `NEPTUNE_API_TOKEN` and `NEPTUNE_PROJECT` in the next cell. Alternatively, you can add the code below to the `decision_trees/train` script before building the Docker image and read the secrets from the `secrets` dictionary instead of `os.envir`.\n",
```